domains/client/domain-operator/src/tests.rs

```diff
@@ -82,6 +82,9 @@ async fn test_domain_instance_bootstrapper() {
 
     assert_eq!(expected_genesis_state_root, genesis_state_root);
 
+    // trigger bundle production
+    alice.send_remark_extrinsic().await.unwrap();
+
     produce_blocks!(ferdie, alice, 3)
```

Comment:
In a lot of tests, we are producing multiple blocks at once, and transactions of imported blocks will be removed from the tx pool asynchronously by the `txpool-notifications` worker, that means there is a race condition that if the `remark` transaction removed from the tx pool after importing the first block then the operator may start skip producing bundle thus cause the test stuck.

I guess this is also the reason `test_domain_tx_propagate` failed since it use `produce_blocks_until` which will keep producing block until the given condition meet.

I suggest moving `alice.send_remark_extrinsic().await` to the `produce_blocks`, `produce_block_with` and `produce_blocks_until` macro to ensure there will be one extrinsic for one block.